spark-ui/src/reducers/SqlReducer.ts

```diff
@@ -485,3 +487,75 @@ function calcBroadcastExchangeDuration(
   }
   return undefined;
 }
+
+function updateNodeMetrics(
+  node: EnrichedSqlNode,
+  metrics: EnrichedSqlMetric[],
+  graph: Graph,
+  allNodes: EnrichedSqlNode[],
+): EnrichedSqlMetric[] {
+
+  const updatedMetrics = calcNodeMetrics(node.type, metrics);
+  addFilterRatioMetric(node, updatedMetrics, graph, allNodes);
+  return updatedMetrics;
+}
+
+function addFilterRatioMetric(
+  node: EnrichedSqlNode,
+  updatedMetrics: EnrichedSqlMetric[],
+  graph: Graph,
+  allNodes: EnrichedSqlNode[],
+): EnrichedSqlMetric[] {
+
+  if (node.nodeName.includes("Filter") || node.nodeName.includes("Join")) {
+    const inputEdges = graph.inEdges(node.nodeId.toString());
+
+    if (!inputEdges || inputEdges.length === 0) {
+      return updatedMetrics;
+    }
+    
+    let totalInputRows = 0;
+    let validPredecessors = 0;
+
+    inputEdges.forEach((edge) => {
+      const inputNode = allNodes.find((n) => n.nodeId.toString() === edge.v);
+      if (inputNode) {
+        const inputRowsMetric = inputNode.metrics?.find((m) =>
+          m.name.includes("rows")
+        );
+        if (inputRowsMetric) {
+          const inputRows = parseFloat(inputRowsMetric.value.replace(/,/g, ""));
+          if (!isNaN(inputRows)) {
+            totalInputRows += inputRows;
+            validPredecessors++;
+          }
+        }
+      }
+    });
+
+    if (validPredecessors === 0) {
+      return updatedMetrics;
+    }
+
+    const outputRowsMetric = updatedMetrics.find((m) => m.name.includes("rows"));
+    if (!outputRowsMetric) {
+      return updatedMetrics;
+    }
+
+    const outputRows = parseFloat(outputRowsMetric.value.replace(/,/g, ""));
+    if (isNaN(outputRows)) {
+      return updatedMetrics;
+    }
+
+    const filterRatio = ((totalInputRows - outputRows) / totalInputRows) * 100;
```

Comment:
use formatPercentage util method

Reply:
Also "filter ratio" is the opposite I think, if you filter 1000 from 1M it's 0.1% filtered not 99%. Maybe the naming should be better here like "filtered rows percentage"

Reply:
I didn't find formatPercentage util I can use in typescript but found NumberFormat, hope it is what you intended :)